dvc/ignore.py

```diff
@@ -72,36 +60,41 @@ def __call__(self, root, dirs, files):
 
 
 class DvcIgnoreFilter(object):
-    def __init__(self, wdir, ignore_file_handler=None):
+    def __init__(self, wdir):
         self.ignores = [
             DvcIgnoreDir(".git"),
             DvcIgnoreDir(".hg"),
             DvcIgnoreDir(".dvc"),
             DvcIgnoreFile(".dvcignore"),
         ]
 
-        self.ignore_file_handler = ignore_file_handler
+        from dvc.repo import Repo
+
+        try:
+            self.tree = WorkingTree(Repo.find_root())
+        except NotDvcRepoError:
+            self.tree = None
```

Comment:
Besides some test, I don't think we could expect that it will be used outside of repo.  I guess its better to expect user to use `dvc_walk` in dvc_repo, than try to forcefully pass tests that did not use dvc_repo.

Reply:
But if we don't make a tree, no `.dvcignore` files would be applied, because `.update()` is skipped. Is this intended?

Reply:
If we are talking about dvcignore itself, then it can't be outside of dvc repo. But keep in mind that we support external directories as dependencies and outputs (e.g. `dvc add /path/to/some/external/dir`), so some methods like `dvc walk` should be able to work there.

Reply:
@efiop thats a good point, I guess dvc_walk should be able to handle out of repository directories.
@Suor That is my mistake. I guess we should proceed with tree, if it exists, and if not, we should just go wit `os.path`. What do you think?